lib/steps/step_list.go

```diff
@@ -48,3 +55,34 @@ func (stepList *StepList) Prepend(step Step) {
 func (stepList *StepList) PrependList(otherList StepList) {
 	stepList.List = append(otherList.List, stepList.List...)
 }
+
+// WrapOptions represents the options given to Wrap.
+type WrapOptions struct {
+	RunInGitRoot     bool
+	StashOpenChanges bool
+}
+
+// Wrap wraps the given StepList in steps that
+// change to the Git root directory or stash away open changes.
+func (stepList *StepList) Wrap(options WrapOptions) {
```

Comment:
Things that my proposed architecture does better:
- user can not manually mess with the internal state of the steplist. Like, for example add steps manually after `Wrap` has been called, or add invalid steps manually 
- it is possible to add steps, get the wrapped steps, and later add more steps and get the wrapped steps again
- it is easier to use in our actual use case, where we request the wrapped steps only once and not repeatedly. In this case, the results can be requested with a single command instead of two:
  - current API: call `Wrap`, then read `steps` to get the wrapped steps
  - proposed API: call `Steps` to get the wrapped steps
- the `WrapOptions` helper struct goes away

I think the proposed API is just more robust design. It gives the user more flexibility while preventing a whole array of possibilities to accidentally misuse this class. 

If we don't need wrapped steps, we can simply provide `stashOpenChanges: false` in the constructor, and then `Steps` doesn't add that wrapper. 

Having said that, if you feel strongly about this, you can ship this without and we can file an issue for this.